src/app/CASESessionManager.cpp

```diff
@@ -54,6 +55,14 @@ void CASESessionManager::FindOrEstablishSession(const ScopedNodeId & peerId, Cal
             {
                 onFailure->mCall(onFailure->mContext, peerId, CHIP_ERROR_NO_MEMORY);
             }
+
+            if (onExtendedConnectionFailure != nullptr)
```

Comment:
I don't think we should have a way to provide both onFailure and onExtendedConnectionFailure.  One or the other, but not both.